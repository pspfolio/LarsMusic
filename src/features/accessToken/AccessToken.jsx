import React, { Component } from 'react';
import { connect } from 'react-redux';
import queryString from 'query-string';
import { setAccessToken } from './accessTokenActions';

export class AccessToken extends Component {
  componentDidMount() {
    const { search } = this.props.location;
    const parsed = queryString.parse(search);
    this.props.setAccessToken(parsed);
  }

  render() {
    return <h1>Redirecting</h1>;
  }
}

const mapDispatchToProps = dispatch => {
  return {
    setAccessToken: accessToken => dispatch(setAccessToken(accessToken))
  };
};

export default connect(null, mapDispatchToProps)(AccessToken);
