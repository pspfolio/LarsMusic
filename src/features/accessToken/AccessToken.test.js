import React from 'react';
import { shallow } from 'enzyme';
import { AccessToken } from './AccessToken';

const location = { search: 'yolo123' };

describe('Accesstoken component', () => {
  it('renders without crashing', () => {
    shallow(<AccessToken setAccessToken={() => {}} location={location} />);
  });

  it('AccessToken renders correctly', () => {
    const wrapper = shallow(<AccessToken setAccessToken={() => {}} location={location} />);
    expect(wrapper).toMatchSnapshot();
  });
});
