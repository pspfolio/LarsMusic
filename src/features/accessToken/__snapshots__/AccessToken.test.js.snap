// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accesstoken component AccessToken renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccessToken
    location={
        Object {
            "search": "yolo123",
          }
    }
    setAccessToken={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Redirecting",
    },
    "ref": null,
    "rendered": "Redirecting",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Redirecting",
      },
      "ref": null,
      "rendered": "Redirecting",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
